dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganisationResource.java

```diff
@@ -454,6 +517,22 @@ private boolean doesOrganisationExistToUser(Long organisationId, Long userId) {
         return organisation.isApproved() || (organisationUser != null);
     }
 
+    /**
+     * Checks if the given user should know of the existence of the organisation by name
+     * For a user to see an organsation, either it must be approved or the user must have a role in the organisation
+     * @param organisationName
+     * @param userId
+     * @return True if organisation exists to user, false otherwise
+     */
+    private boolean doesOrganisationByNameExistToUser(String organisationName, Long userId) {
+        Organisation organisation = organisationDAO.findByName(organisationName);
+        if (organisation == null) {
+            return false;
+        }
+        OrganisationUser organisationUser = getUserOrgRole(organisation, userId);
```

Comment:
If the organization is approved, you could save yourself the cost of looking up the org user, although it looks like it is pretty cheap, so not a big savings.